docs/jj_first_principles.md

```diff
@@ -0,0 +1,50 @@
+# Jujutsu from first principles
+
+## Preface
+
+Why does Jujutsu exist and which problems does it solve? This document tries to
+answer both of these questions while expanding on the design in a user-friendly
+way. 
+
+At its core Jujutsu is [Version Control System][vcs] which scales to huge 
+repositories at [Google scale][billion-lines]. Many design choices are 
+influenced by the concurrent commits happening in Googles Monorepo, as there 
+are always multiple people working on the same file(s) at the same time.
+
+## Core Tenets
+
+Jujutsu's core tenets are:
```

Comment:
Interesting heading. I haven't really thought about what I which bullets I would put here. Thinking about it now, here's what I can think of:

* User-friendliness: Making the working copy a commit is simpler. This is how the project started.
* All operations must be able to scale at least to Google-scale repos (lots of commits, lots of files): Laziness is important, must avoid accessing data unnecessarily
* The "repository" (commit graph) is the source of truth: The working copy is just one way of editing commits
* Git-interop: Git is everywhere. We need to have good interop to gain adoption.
* Pluggable storage: Must be easy to integrate with different commit storage, virtual file systems, etc.

Reply:
I've added them.